src/engine/N3Base/N3UIList.cpp

```diff
@@ -195,6 +195,17 @@ bool CN3UIList::SetString(int iIndex, const std::string & szString) {
     return false;
 }
 
+CN3UIString * CN3UIList::GetChildStrFromList(std::string str) {
+    for (std::list<CN3UIString *>::iterator it = m_ListString.begin(); it != m_ListString.end(); ++it) {
+        CN3UIString * pUIString = (*it);
+        if (pUIString->GetString() == str) {
+            return pUIString;
+        }
+    }
```

Comment:
Use C++'s range-based for loops (aka foreach in other languages): https://en.cppreference.com/w/cpp/language/range-for